cinch/playbooks/jenkins_restart_utility.yml

```diff
@@ -0,0 +1,139 @@
+# This playbook aims to gather diagnostics data in the event of
+# trouble with Jenkins, and then restarts Jenkins to keep operations running.
+# By default, only the diagnostics data is collected, but the Jenkins service
+# restart can be enabled by setting the 'restart_jenkins' variable to true:
+
+# Example:
+# ansible-playbook -u root --private-key /path/to/key
+# -i '10.0.183.169', jenkins_restart_utility.yml -e "restart_jenkins=true"
+
+- hosts: all
+  vars:
+    jenkins_pid_path: '/run/jenkins.pid'
+  tasks:
+    # This block ignores all errors to ensure that Jenkins is restarted on
+    # demand even if we can't gather diagnostics data.
+    - name: gather diagnostics data
+      block:
+        - name: check if Jenkins is running
+          stat:
```

Comment:
Indeed there is not a pretty solution here, even with systemd.  At least for the case of the Jenkins systemd service, there's not an easily parseable way to get the last PID without some serious shell nonsense that I'd like to avoid.  See the output of these commands if you're curious:

```
systemctl show jenkins --no-pager
systemctl status -l jenkins --no-pager
```

In the end, we just need to *try* to grab the diagnostics data, and we need the PID to do that, and if we can't get that data, then at least we can restart the service for the user.  The only other alternative I can think of here is to use something like `pgrep -f jenkins`, but what happens if some other process has **jenkins** in the name there?  Not great either.  I think that using **/run/jenkins.pid** as the best source of truth to grab the diagnostics data will have to be sufficient, and per @greg-hellings suggestion I will rename the task to better indicate that we're not definitively making a decision that Jenkins is actually running or not.